### PR TITLE
(GH-677) set default for WebLinkRoot as RepositoryName instead of title.

### LINKED ISSUE #677: Publishing docs to github-pages does not work, when title and repositoryName do not match
When publishing to github pages, the url of the "website" is generated in the following schema: `[repo-owner].github.io/[repo-name]`. However, Cake.Recipe assumes an url in the form of `[repo-owner].github.io/[addin-title]`. This leads to errors, when the repository-name does not match the addin-title.

As an example, I (foolishly) created Cake.AsciiDoctorJ at https://cake-contrib.github.io/cake.asciidoctorj/ (note the lowercase repository-name).

So my `recipe.cake` looks like:

```cs
    // ...
    title: "Cake.AsciiDoctorJ",
    repositoryName: "cake.asciidoctorj",
    // ...
```

This leads, in the default settings to a wyam generated code of:

```html
<header class="main-header">   
    <a href="/Cake.AsciiDoctorJ/" class="logo">
        <span>Cake.AsciiDoctorJ</span>
    </a>   
    <!-- more stuff -->                     
</header>
```

which simply does not work.

So, modifying the `recipe.cake` to:

```cs
    // ...
    title: "Cake.AsciiDoctorJ",
    repositoryName: "cake.asciidoctorj",
    webLinkRoot: "cake.asciidoctorj",
    // ...
```

will fix this.

However I feel this is a bug in Cake.Recipe. It *could* work without setting `webLinkRoot` explicitly since both `title` and `repositoryName` are known.

### PR DESCRIPTION
**Pull request by nils-a**
fixes #677

### CODE REVIEW

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@nils-a your changes have been merged, thanks for your contribution 👍

File: Cake.Recipe/Content/parameters.cake [hunk 1]
```diff
@@ -459,7 +459,7 @@ public static class BuildParameters
         WyamTheme = wyamTheme ?? "Samson";
         WyamSourceFiles = wyamSourceFiles ?? "../../" + SourceDirectoryPath.FullPath + "/**/{!bin,!obj,!packages,!*.Tests,}/**/*.cs";
         WebHost = webHost ?? string.Format("{0}.github.io", repositoryOwner);
-        WebLinkRoot = webLinkRoot ?? title;
+        WebLinkRoot = webLinkRoot ?? RepositoryName;
```
Comment (AdmiringWorm) [inline Cake.Recipe/Content/parameters.cake:462]: Just a thought, shouldn't we perhaps have the fallback being set to CI provider repository name instead, and if that is null fall all the way back to the `RepositoryName` variable.

For instance:

```suggestion
        WebLinkRoot = webLinkRoot ?? BuildProvider.Repository?.Name ?? RepositoryName;
```

Comment (gep13) [reply Cake.Recipe/Content/parameters.cake:462]: Hmm, now that we have this information, that I a good suggestion.  @nils-a what do you think?  If you agree, would you mind putting in a PR to adjust?

Comment (nils-a) [reply Cake.Recipe/Content/parameters.cake:462]: Sure, but.. I don't think this is "good" :-)

I *think* it should be 

```suggestion
        RepositoryName = repositoryName ?? BuildProvider.Repository?.Name ?? Title;
        WebLinkRoot = webLinkRoot ?? RepositoryName;
```

(this does not work so goog using the *suggestions* - `RespositoryName` should be modified in that case, should it not?)

Comment (gep13) [reply Cake.Recipe/Content/parameters.cake:462]: @nils-a I think that this is a sensible suggestion.  @AdmiringWorm thoughts?  In either case, I was thinking about this change, and I think we need to update the docs: https://cake-contrib.github.io/Cake.Recipe/docs/fundamentals/set-parameters#weblinkroot @nils-a would you be in a position to update this?

Comment (nils-a) [reply Cake.Recipe/Content/parameters.cake:462]: will do.

**Pull request merged**